File: solutions/zig/armstrong-numbers/1/armstrong_numbers.zig
pub fn isArmstrongNumber(num: u128) bool {
    var n = num;
    var l: u128 = 0;
    if (n == 0) {
        l = 1;
    } else {
        while (n != 0) {
            l += 1;
            n /= 10;
        }
    }
    n = num;
    var ans: u128 = 0;
    while (n != 0) {
        ans += pow(n % 10, l);
        n /= 10;
    }
    return ans == num;
}

pub fn pow(a: u128, b: u128) u128 {
    if (b == 0) {
        return 1;
    } else {
        const p = pow(a, b / 2);
        var ans = p * p;
        if (b & 1 == 1) {
            ans *= a;
        }
        return ans;
    }
}

File: solutions/csharp/armstrong-numbers/1/ArmstrongNumbers.cs
public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        if (number < 0)
        {
            return false;
        }
        var n = number;
        var digits = new List<int>();
        var l = 0;
        do
        {
            digits.Add(n % 10);
            n /= 10;
            l += 1;
        } while (n != 0);
        var ans = 0;
        foreach (var d in digits)
        {
            ans += (int)Math.Pow(d, l);
        }
        return ans == number;
    }
}